### PR TITLE
Handle SIGWINCH signal

### PR DESCRIPTION
**Pull request by lpereira**
These changes will make kilo properly handle the SIGWINCH signal, so that if the window is resized, the editing area is also updated.

Some compile warnings have been removed as well.

### CODE REVIEW

File: kilo.c [hunk 1]
```diff
@@ -36,20 +36,23 @@
...
 #include <errno.h>
-#include <string.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdarg.h>
+#include <stdio.h>
 #include <stdlib.h>
-#include <ctype.h>
-#include <sys/types.h>
+#include <stdlib.h>
```
Comment (vsimon) [inline kilo.c]: This is including stdlib.h twice.

Comment (lpereira) [inline kilo.c]: It was already like this, but I've updated the PR. Thanks.

**Comment by kristofer** [comment]
+1 on sigwinch

**Comment by antirez** [comment]
Thanks, merged with a few changes to make it compile on OSX and don't collied with other PRs I already merged.

**Pull request closed without merging**